test/server/cards/10-TEL/ChampionsOfYomi.spec.js

```diff
@@ -0,0 +1,167 @@
+                    player1: {
+                        inPlay: ['champions-of-yomi'],
+                        hand: ['fine-katana']
+                    },
+                    player2: {}
+                });
+                this.fineKatana = this.player1.findCardByName('fine-katana');
+                this.championsOfYomi = this.player1.findCardByName('champions-of-yomi');
+            });
+
+            it('should not allow non-Spirit attachments', function() {
+                this.player1.clickCard(this.fineKatana);
+                expect(this.player1).not.toBeAbleToSelect(this.championsOfYomi);
+            });
+
+            it('should remain in play at the end of the phase', function() {
+                this.nextPhase();
+                expect(this.championsOfYomi.location).toBe('play area');
+            });
+        });
+
+        describe('when in attacking player\'s discard and the conflict resolves', function() {
+            beforeEach(function() {
+                this.setupTest({
+                    phase: 'conflict',
+                    player1: {
+                        stronghold: 'yojin-no-shiro',
+                        inPlay: ['miwaku-kabe-guard'],
+                        hand: ['spreading-the-darkness'],
+                        dynastyDiscard: ['champions-of-yomi']
+                    },
+                    player2: {
+                        fate: 20,
+                        inPlay: ['kakita-toshimoko', 'yoritomo'],
+                        hand: ['assassination', 'fallen-in-battle']
+                    }
+                });
+                this.yojinNoShiro = this.player1.findCardByName('yojin-no-shiro');
+                this.championsOfYomi = this.player1.findCardByName('champions-of-yomi', 'dynasty discard pile');
+                this.miwakuKabeGuard = this.player1.findCardByName('miwaku-kabe-guard');
+                this.spreadingTheDarkness = this.player1.findCardByName('spreading-the-darkness');
+                this.kakitaToshimoko = this.player2.findCardByName('kakita-toshimoko');
+                this.yoritomo = this.player2.findCardByName('yoritomo');
+                this.assassination = this.player2.findCardByName('assassination');
+                this.fallenInBattle = this.player2.findCardByName('fallen-in-battle');
+                this.noMoreActions();
+                this.initiateConflict({
+                    ring: 'water',
+                    attackers: [this.miwakuKabeGuard],
+                    defenders: [this.kakitaToshimoko]
+                });
+            });
+
+            it('should be put into play when triggered after losing', function() {
+                this.noMoreActions();
+                expect(this.player1).toBeAbleToSelect(this.championsOfYomi);
+                this.player1.clickCard(this.championsOfYomi);
+                expect(this.player1).toBeAbleToSelect(this.yojinNoShiro);
+                this.player1.clickCard(this.yojinNoShiro);
```

Comment:
after this you should check that the cost of the stronghold being bowed was paid.

Reply:
Good call, done.